### PR TITLE
postgresql: some improvements to the derivation

### PR DESCRIPTION
**Pull request by wolfgangwalther**
## Description of changes

This fell out of my work on "building with meson / in pkgsStatic", which I have recently been progressing well with. Still nothing to show, yet, though.

@Ma27 let me know what you think!

## Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [x] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [x] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#linking-nixos-module-tests-to-a-package) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [24.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2411.section.md) (or backporting [23.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) and [24.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2405.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

### CODE REVIEW

**Review by doronbehar — APPROVED** [review]
Changes look very good and are well documented. Should be good to go once CI is green.

**Comment by Ma27** [comment]
Sure, will do so tomorrow.

**Review by doronbehar — APPROVED** [review]

**Comment by doronbehar** [comment]
Oh but there is a merge conflict.

**Comment by doronbehar** [comment]
Could you please verify that on the rebased branch everything works at least on your platform? ofborg's logs are not green and it is not clear to me why.

**Comment by doronbehar** [comment]
> Imho, ofborg had been failing before, too. I did verify the changes on master only, so far. I don't expect any change after the kind of conflicts I fixed.
> 
> I will kick off another build on staging though. Might take a while ;)

Oh I see, I haven't noticed we are targeting staging, so there is no need for that. In fact, I think this PR is mature enough to be further cooked in staging :).

**Comment by doronbehar** [comment]
Aish I would have merged this right now if you hadn't forced pushed. Let's merge this after evaluation is green.

File: pkgs/servers/sql/postgresql/generic.nix [hunk 1]
```diff
@@ -102,16 +98,17 @@ let
...
       ++ lib.optionals jitSupport [ llvmPackages.llvm ]
       ++ lib.optionals lz4Enabled [ lz4 ]
       ++ lib.optionals zstdEnabled [ zstd ]
-      ++ lib.optionals systemdSupport' [ systemdLibs ]
+      ++ lib.optionals systemdSupport [ systemdLibs ]
       ++ lib.optionals pythonSupport [ python3 ]
       ++ lib.optionals gssSupport [ libkrb5 ]
       ++ lib.optionals stdenv'.hostPlatform.isLinux [ linux-pam ]
-      ++ lib.optionals (!stdenv'.hostPlatform.isDarwin) [ libossp_uuid ];
+      ++ lib.optionals (!stdenv'.hostPlatform.isDarwin) [ e2fsprogs ];
```
Comment (Ma27) [inline pkgs/servers/sql/postgresql/generic.nix:111]: Can you explain why both `libossp_uuid` & `e2fsprogs` are needed? From what I've read in the commit message, only one of them is required?

Comment (Ma27) [reply pkgs/servers/sql/postgresql/generic.nix:111]: > libossp_uuid is removed, but you mean libuuid (further up) and e2fsprogs, I guess.

Correct, good morning ma27 🙃

**Review by Ma27 — APPROVED** [review]
Thank _you_!

**Pull request merged**